tests/test_lane.py

```diff
@@ -1092,57 +1092,77 @@ def test_only_show_ready_deliverable_works(self):
         # w1 has licenses but no available copies. It's available
         # unless site policy is to hide books like this.
         w1 = self._work(with_license_pool=True)
+        w1.presentation_edition.title = 'I have no available copies'
         w1.license_pools[0].open_access = False
         w1.license_pools[0].licenses_owned = 10
         w1.license_pools[0].licenses_available = 0
 
         # w2 has no delivery mechanisms.
         w2 = self._work(with_license_pool=True, with_open_access_download=False)
+        w2.presentation_edition.title = 'I have no delivery mechanisms'
         for dm in w2.license_pools[0].delivery_mechanisms:
             self._db.delete(dm)
 
         # w3 is not presentation ready.
         w3 = self._work(with_license_pool=True)
+        w3.presentation_edition.title = "I'm not presentation ready"
         w3.presentation_ready = False
 
         # w4's only license pool is suppressed.
         w4 = self._work(with_open_access_download=True)
+        w4.presentation_edition.title = "I am suppressed"
         w4.license_pools[0].suppressed = True
 
         # w5 has no licenses.
         w5 = self._work(with_license_pool=True)
+        w5.presentation_edition.title = "I have no owned licenses."
         w5.license_pools[0].open_access = False
         w5.license_pools[0].licenses_owned = 0
 
         # w6 is an open-access book, so it's available even though
         # licenses_owned and licenses_available are zero.
         w6 = self._work(with_open_access_download=True)
+        w6.presentation_edition.title = "I'm open-access."
         w6.license_pools[0].open_access = True
         w6.license_pools[0].licenses_owned = 0
         w6.license_pools[0].licenses_available = 0
 
         # w7 is not open-access. We own licenses for it, and there are
         # licenses available right now. It's available.
         w7 = self._work(with_license_pool=True)
+        w7.presentation_edition.title = "I have available licenses."
         w7.license_pools[0].open_access = False
         w7.license_pools[0].licenses_owned = 9
         w7.license_pools[0].licenses_available = 5
 
+        # w8 has a delivery mechanism that can't be rendered by the
+        # default client.
+        w8 = self._work(with_license_pool=True)
+        w8.presentation_edition.title = "I have a weird delivery mechanism"
+        [pool] = w8.license_pools
+        for dm in pool.delivery_mechanisms:
+            self._db.delete(dm)
+        self._db.commit()
+        pool.set_delivery_mechanism(
+            "weird content type", "weird DRM scheme", "weird rights URI",
+            None
+        )
+        
```

Comment:
This is the code that would have shown our "only delivery mechanisms the default client can render" code was broken.